src/Symfony/Component/Routing/Loader/XmlFileLoader.php

```diff
@@ -133,6 +133,22 @@ protected function parseRoute(RouteCollection $collection, \DOMElement $node, $p
 
         list($defaults, $requirements, $options, $condition) = $this->parseConfigs($node, $path);
 
+        if (isset($requirements['_method'])) {
+            if (0 === count($methods)) {
+                $methods = explode('|', $requirements['_method']);
+            }
+            unset($requirements['_method']);
+            trigger_error(sprintf('The "_method" requirement in file "%s" is deprecated since version 2.2 and will be removed in 3.0. Use the methods attribute instead.', $path), E_USER_DEPRECATED);
```

Comment:
we could even add the route name in the message here as we have it in ``$id``, making the lookup even faster